api/src/main/java/org/openmrs/patient/impl/BaseHyphenatedIdentifierValidator.java

```diff
@@ -54,8 +54,15 @@ public String getValidIdentifier(String undecoratedIdentifier) throws UnallowedI
 	@Override
 	public boolean isValid(String identifier) throws UnallowedIdentifierException {
 		
-		if (identifier.indexOf("-") < 1) {
-			throw new UnallowedIdentifierException("Identifier must contain something besides the check digit.");
+		if (identifier.indexOf("-") <= 0) {
```

Comment:
Does <= 0 make it any better than < 1?

Reply:
Sorry, I think I changed it to <0, then realised that it should fail on zero, so I added an "=" sign. I should have reinstated <1.